pkg/config/execution_unit.go

```diff
@@ -46,15 +46,15 @@ type (
 
 	// KubernetesLimits represents the configurability of kubernetes limits for execution units which match the kubernetes compatibility
 	KubernetesLimits struct {
-		// Cpu specifies the limit per pod in millicores
-		Cpu int `json:"cpu,omitempty" yaml:"cpu,omitempty" toml:"cpu,omitempty"`
+		// Cpu specifies the limit per pod in millicores. It is "any" so that the user can specify it as either a string or a number
+		Cpu any `json:"cpu,omitempty" yaml:"cpu,omitempty" toml:"cpu,omitempty"`
 		// Memory specifies the limit per pod in MB
-		Memory int `json:"memory,omitempty" yaml:"memory,omitempty" toml:"memory,omitempty"`
+		Memory any `json:"memory,omitempty" yaml:"memory,omitempty" toml:"memory,omitempty"`
```

Comment:
why any instead of int?

Reply:
Whoops, forgot to copy the comments from a couple lines above. But it's so that they can do either an int or string. For example, `memory: 128M` is valid, and parses to a string. But you have to allow numbers too, because `memory: 128` is also valid and parses to int.

Reply:
im wondering if we force either a string or an int though to make it easier for us, theyll still be able to understand the values

Reply:
Imo, given that all we do here is pass the value through to k8s (which can handle it as both string and int), the overriding concern should be the user experience — which seems significantly better with this as an `any`. My understanding is that this is used to read their input, right? So it's not enough that the user can understand it, they also have to be able to input the values they want: whether that's `cpu: 1` (an int), `cpu: 0.5` (a float),  or `cpu: 100m` (a string).

Unless I keep these as `any`, various `Test_configureContainer` cases fail.

Reply:
alright, as long as that doesnt break our casting to kubernetes type params from the generic params then thats fine

Reply:
Ah, I see. Yeah it doesn't matter for the casting: our method is to json-ify the generic InfraParams and then de-json-ify that back as KubernetesTypesParams, so `any` works totally fine (and is in fact the type of the incoming InfraParams, which is an `InfraParams map[string]any`).